packages/geolocator/tizen/src/permission_manager.cc

```diff
@@ -1,97 +1,87 @@
-// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
 #include "permission_manager.h"
 
+#ifndef TV_PROFILE
 #include <Ecore.h>
 #include <privacy_privilege_manager.h>
-
-#include <vector>
+#include <tizen.h>
+#endif
 
 #include "log.h"
 
-namespace {
-
-constexpr char kPrivilegeLocation[] = "http://tizen.org/privilege/location";
-
-struct PermissionResponse {
-  ppm_call_cause_e cause;
-  ppm_request_result_e result;
-  bool received = false;
-};
-
-}  // namespace
-
-PermissionManager::PermissionManager() {}
-PermissionManager::~PermissionManager() {}
-
-TizenResult PermissionManager::CheckPermissionStatus(
-    PermissionStatus *permission_status) {
-  ppm_check_result_e check_result;
-
-  int result = ppm_check_permission(kPrivilegeLocation, &check_result);
-  if (result != PRIVACY_PRIVILEGE_MANAGER_ERROR_NONE) {
-    return TizenResult(result);
+PermissionStatus PermissionManager::CheckPermission(
+    const std::string &privilege) {
+#ifdef TV_PROFILE
+  return PermissionStatus::kAllow;
+#else
+  ppm_check_result_e result;
+  int ret = ppm_check_permission(privilege.c_str(), &result);
+  if (ret != PRIVACY_PRIVILEGE_MANAGER_ERROR_NONE) {
+    LOG_ERROR("Permission check failed [%s]: %s", privilege.c_str(),
+              get_error_message(ret));
+    return PermissionStatus::kError;
   }
 
-  switch (check_result) {
-    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_DENY:
-    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK:
-      *permission_status = PermissionStatus::kDenied;
-      break;
+  switch (result) {
     case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ALLOW:
+      return PermissionStatus::kAllow;
+    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_ASK:
+      return PermissionStatus::kAsk;
+    case PRIVACY_PRIVILEGE_MANAGER_CHECK_RESULT_DENY:
     default:
-      *permission_status = PermissionStatus::kAlways;
-      break;
+      return PermissionStatus::kDeny;
   }
-  return TizenResult();
+#endif
 }
 
-void PermissionManager::RequestPermssion(const OnSuccess &on_success,
-                                         const OnFailure &on_failure) {
-  const char *permission = kPrivilegeLocation;
-  PermissionResponse response;
+PermissionResult PermissionManager::RequestPermssion(
```

Comment:
```suggestion
PermissionResult PermissionManager::RequestPermission(
```
This is my mistake. I'll update image_picker and permission_handler as well.